### PR TITLE
Add docstyle rule to the linter and subsequent refactor.

### PR DESCRIPTION
**Pull request by thanacles**
Associated to: https://github.com/quantumlib/Cirq/issues/3388

Added the docstyle linter and refactored trivial pydocs.  If the refactor wasn't trivial I just disable the rule for the pydoc.

We'll need to add the docparams plugin in a later PR.

### CODE REVIEW

File: examples/qubit_characterizations_example.py [hunk 1]
```diff
@@ -3,8 +3,7 @@
 
 
 def main(minimum_cliffords=5, maximum_cliffords=20, cliffords_step=5):
-    """
-    Examples of how to run various qubit characterizations.  This example
+    """Examples of how to run various qubit characterizations.  This example
```
Comment (balopat) [inline examples/qubit_characterizations_example.py:6]: Fix summary.

File: examples/quantum_fourier_transform.py [hunk 2]
```diff
@@ -1,5 +1,4 @@
-"""
-Creates and simulates a circuit for Quantum Fourier Transform(QFT)
+"""Creates and simulates a circuit for Quantum Fourier Transform(QFT)
```
Comment (balopat) [inline examples/quantum_fourier_transform.py:1]: fix summary

File: examples/hhl.py [hunk 3]
```diff
@@ -274,8 +268,7 @@ def simulate(circuit):
 
 
 def main():
-    """
-    Simulates HHL with matrix input, and outputs Pauli observables of the
+    """Simulates HHL with matrix input, and outputs Pauli observables of the
```
Comment (balopat) [inline examples/hhl.py:271]: fix summary

File: examples/hhl.py [hunk 4]
```diff
@@ -154,8 +150,7 @@ def _decompose_(self, qubits):
 
 
 class EigenRotation(cirq.Gate):
-    """
-    EigenRotation performs the set of rotation on the ancilla qubit equivalent
+    """EigenRotation performs the set of rotation on the ancilla qubit equivalent
```
Comment (balopat) [inline examples/hhl.py:153]: fix summary

File: examples/hhl.py [hunk 5]
```diff
@@ -1,5 +1,4 @@
-"""
-Demonstrates the algorithm for solving linear systems by Harrow, Hassidim,
+"""Demonstrates the algorithm for solving linear systems by Harrow, Hassidim,
```
Comment (balopat) [inline examples/hhl.py:1]: fix summary line

File: examples/direct_fidelity_estimation.py [hunk 6]
```diff
@@ -347,8 +337,7 @@ def direct_fidelity_estimation(
     n_measured_operators: Optional[int],
     samples_per_term: int,
 ):
-    """
-    Implementation of direct fidelity estimation, as per 'Direct Fidelity
+    """Implementation of direct fidelity estimation, as per 'Direct Fidelity
```
Comment (balopat) [inline examples/direct_fidelity_estimation.py:340]: fix summary line

File: examples/direct_fidelity_estimation.py [hunk 7]
```diff
@@ -321,8 +312,7 @@ class Result:
 
 @dataclass
 class DFEIntermediateResult:
-    """
-    A container for the various debug and run data from calling the function
+    """A container for the various debug and run data from calling the function
```
Comment (balopat) [inline examples/direct_fidelity_estimation.py:315]: fix summary line

File: examples/direct_fidelity_estimation.py [hunk 8]
```diff
@@ -235,8 +230,7 @@ def _estimate_pauli_traces_clifford(
 def _estimate_pauli_traces_general(
     qubits: List[cirq.Qid], circuit: cirq.Circuit, n_measured_operators: Optional[int]
 ) -> List[PauliTrace]:
-    """
-    Estimates the Pauli traces in case the circuit is not Clifford. In this case
+    """Estimates the Pauli traces in case the circuit is not Clifford. In this case
```
Comment (balopat) [inline examples/direct_fidelity_estimation.py:233]: fix summary line

File: examples/direct_fidelity_estimation.py [hunk 9]
```diff
@@ -177,8 +173,7 @@ def _estimate_pauli_traces_clifford(
     stabilizer_basis: List[cirq.DensePauliString],
     n_measured_operators: Optional[int],
 ) -> List[PauliTrace]:
-    """
-    Estimates the Pauli traces in case the circuit is Clifford. When we have a
+    """Estimates the Pauli traces in case the circuit is Clifford. When we have a
```
Comment (balopat) [inline examples/direct_fidelity_estimation.py:176]: fix summary line

File: examples/direct_fidelity_estimation.py [hunk 10]
```diff
@@ -157,8 +154,7 @@ def _enumerate_all_from_stabilizer_bases(
 
 @dataclass
 class PauliTrace:
-    """
-    A class that contains the Pauli states as described on page 2 of:
+    """A class that contains the Pauli states as described on page 2 of:
```
Comment (balopat) [inline examples/direct_fidelity_estimation.py:157]: fix summary line

File: examples/direct_fidelity_estimation.py [hunk 11]
```diff
@@ -133,8 +131,7 @@ def _randomly_sample_from_stabilizer_bases(
 def _enumerate_all_from_stabilizer_bases(
     stabilizer_basis: List[cirq.DensePauliString], n_qubits: int
 ):
-    """
-    Given a stabilizer basis, creates the exhaustive list of Pauli states that
+    """Given a stabilizer basis, creates the exhaustive list of Pauli states that
```
Comment (balopat) [inline examples/direct_fidelity_estimation.py:134]: fix summary line

File: examples/direct_fidelity_estimation.py [hunk 12]
```diff
@@ -105,8 +104,7 @@ async def estimate_characteristic_function(
 def _randomly_sample_from_stabilizer_bases(
     stabilizer_basis: List[cirq.DensePauliString], n_measured_operators: int, n_qubits: int
 ):
-    """
-    Given a stabilizer basis, randomly creates Pauli states by including the
+    """Given a stabilizer basis, randomly creates Pauli states by including the
```
Comment (balopat) [inline examples/direct_fidelity_estimation.py:107]: fix summary line

File: examples/direct_fidelity_estimation.py [hunk 13]
```diff
@@ -75,8 +75,7 @@ async def estimate_characteristic_function(
     sampler: cirq.Sampler,
     samples_per_term: int,
 ):
-    """
-    Estimates the characteristic function using a (noisy) circuit simulator by
+    """Estimates the characteristic function using a (noisy) circuit simulator by
```
Comment (balopat) [inline examples/direct_fidelity_estimation.py:78]: fix summary line

File: examples/bell_inequality.py [hunk 14]
```diff
@@ -1,5 +1,4 @@
-"""
-Bell's theorem or inequality proves that entanglement based
+"""Bell's theorem or inequality proves that entanglement based
```
Comment (balopat) [inline examples/bell_inequality.py:1]: fix summary line

File: dev_tools/incremental_coverage.py [hunk 15]
```diff
@@ -128,8 +127,7 @@ def fix_line_from_coverage_file(line):
 def get_incremental_uncovered_lines(
     abs_path: str, base_commit: str, actual_commit: Optional[str]
 ) -> List[Tuple[int, str, str]]:
-    """
-    Uses git diff and the annotation files created by
+    """Uses git diff and the annotation files created by
```
Comment (balopat) [inline dev_tools/incremental_coverage.py:130]: fix summary line

File: cirq-pasqal/cirq_pasqal/pasqal_sampler_test.py [hunk 16]
```diff
@@ -59,8 +59,7 @@ def test_pasqal_circuit_init():
 @patch('cirq_pasqal.pasqal_sampler.requests.get')
 @patch('cirq_pasqal.pasqal_sampler.requests.post')
 def test_run_sweep(mock_post, mock_get):
-    """
-    Encodes a random binary number in the qubits, sweeps between odd and even
+    """Encodes a random binary number in the qubits, sweeps between odd and even
```
Comment (balopat) [inline cirq-pasqal/cirq_pasqal/pasqal_sampler_test.py:62]: fix summary line

File: cirq-google/cirq_google/optimizers/convert_to_sycamore_gates.py [hunk 17]
```diff
@@ -496,8 +494,7 @@ def rzz(theta: float, q0: cirq.Qid, q1: cirq.Qid) -> cirq.OP_TREE:
 
 
 def cphase(theta: float, q0: cirq.Qid, q1: cirq.Qid) -> cirq.OP_TREE:
-    """
-    Implement a cphase using the Ising gate generated from 2 Sycamore gates
+    """Implement a cphase using the Ising gate generated from 2 Sycamore gates
```
Comment (balopat) [inline cirq-google/cirq_google/optimizers/convert_to_sycamore_gates.py:497]:
```suggestion
    """Implements a cphase using the Ising gate generated from 2 Sycamore gates.
```

File: cirq-google/cirq_google/optimizers/convert_to_sycamore_gates.py [hunk 18]
```diff
@@ -408,8 +407,7 @@ def decompose_swap_into_syc(a: cirq.Qid, b: cirq.Qid):
 
 
 def find_local_equivalents(unitary1: np.ndarray, unitary2: np.ndarray):
-    """
-    Given two unitaries with the same interaction coefficients but different
+    """Given two unitaries with the same interaction coefficients but different
```
Comment (balopat) [inline cirq-google/cirq_google/optimizers/convert_to_sycamore_gates.py:410]: fix summary line

File: cirq-google/cirq_google/optimizers/convert_to_sycamore_gates.py [hunk 19]
```diff
@@ -175,8 +175,7 @@ def known_two_q_operations_to_sycamore_operations(
     op: cirq.Operation,
     tabulation: Optional[GateTabulation] = None,
 ) -> cirq.OP_TREE:
-    """
-    Synthesize a known gate operation to a sycamore operation
+    """Synthesize a known gate operation to a sycamore operation
```
Comment (balopat) [inline cirq-google/cirq_google/optimizers/convert_to_sycamore_gates.py:178]:
```suggestion
    """Synthesizes a known gate operation to a sycamore operation.
```

File: cirq-google/cirq_google/optimizers/convert_to_sycamore_gates.py [hunk 20]
```diff
@@ -99,8 +100,7 @@ def _is_native_sycamore_op(self, op: cirq.Operation) -> bool:
         return False
 
     def _convert_one(self, op: cirq.Operation) -> cirq.OP_TREE:
-        """
-        Decomposer intercept:  Upon cirq.protocols.decompose catch and
+        """Decomposer intercept:  Upon cirq.protocols.decompose catch and
```
Comment (balopat) [inline cirq-google/cirq_google/optimizers/convert_to_sycamore_gates.py:103]: fix summary line

File: cirq-google/cirq_google/optimizers/convert_to_sqrt_iswap.py [hunk 21]
```diff
@@ -55,8 +56,7 @@ def __init__(self, ignore_failures=False) -> None:
         self.ignore_failures = ignore_failures
 
     def _convert_one(self, op: cirq.Operation) -> cirq.OP_TREE:
-        """
```
Comment (balopat) [inline cirq-google/cirq_google/optimizers/convert_to_sqrt_iswap.py:58]: fix summary line

File: cirq-google/cirq_google/engine/client/quantum_v1alpha1/gapic/enums.py [hunk 22]
```diff
@@ -21,8 +21,7 @@
```
Comment (balopat) [inline cirq-google/cirq_google/engine/client/quantum_v1alpha1/gapic/enums.py:21]: let's turn off pylint for this full file, I think this is generated.

File: cirq-core/cirq/study/resolver_test.py [hunk 23]
```diff
@@ -256,8 +255,7 @@ def _resolved_value_(self):
 
 
 def test_compose():
-    """
-    Calling cirq.resolve_paramters on a ParamResolver composes that resolver
+    """Calling cirq.resolve_paramters on a ParamResolver composes that resolver
```
Comment (balopat) [inline cirq-core/cirq/study/resolver_test.py:258]: fix summary line

File: cirq-core/cirq/study/resolver_test.py [hunk 24]
```diff
@@ -143,8 +143,7 @@ def test_param_dict_iter():
 
 
 def test_formulas_in_param_dict():
-    """
-    Param dicts are allowed to have str or sympy.Symbol as keys and
+    """Param dicts are allowed to have str or sympy.Symbol as keys and
```
Comment (balopat) [inline cirq-core/cirq/study/resolver_test.py:146]: fix summary line

File: cirq-core/cirq/sim/sparse_simulator_test.py [hunk 25]
```diff
@@ -1175,8 +1175,7 @@ def test_random_seed_mixture_deterministic():
 
 
 def test_entangled_reset_does_not_break_randomness():
-    """
-    A previous version of cirq made the mistake of assuming that it was okay to
+    """A previous version of cirq made the mistake of assuming that it was okay to
```
Comment (balopat) [inline cirq-core/cirq/sim/sparse_simulator_test.py:1178]: fix summary line

File: cirq-core/cirq/protocols/json_serialization.py [hunk 26]
```diff
@@ -171,8 +171,7 @@ def json_serializable_dataclass(
     unsafe_hash: bool = False,
     frozen: bool = False,
 ):
-    """
-    Create a dataclass that supports JSON serialization
+    """Create a dataclass that supports JSON serialization
```
Comment (balopat) [inline cirq-core/cirq/protocols/json_serialization.py:174]:
```suggestion
    """Create a dataclass that supports JSON serialization.
```

File: cirq-core/cirq/protocols/has_stabilizer_effect_protocol.py [hunk 27]
```diff
@@ -23,8 +23,7 @@
 
 
 def has_stabilizer_effect(val: Any) -> bool:
-    """
-    Returns whether the input has a stabilizer effect.
+    """Returns whether the input has a stabilizer effect.
```
Comment (balopat) [inline cirq-core/cirq/protocols/has_stabilizer_effect_protocol.py:26]:
```suggestion
    """Returns whether the input has a stabilizer effect.
    
```

File: cirq-core/cirq/ops/fsim_gate.py [hunk 28]
```diff
@@ -79,7 +79,8 @@ class FSimGate(gate_features.TwoQubitGate, gate_features.InterchangeableQubitsGa
     """
 
     def __init__(self, theta: float, phi: float) -> None:
-        """
+        """inits FSimGate.
```
Comment (balopat) [inline cirq-core/cirq/ops/fsim_gate.py:82]:
```suggestion
        """Inits FSimGate.
```

File: cirq-core/cirq/neutral_atoms/neutral_atom_devices.py [hunk 29]
```diff
@@ -264,8 +257,7 @@ def _are_qubit_lists_too_close(self, *qubit_lists: Iterable[raw_types.Qid]) -> b
         )
 
     def can_add_operation_into_moment(self, operation: ops.Operation, moment: ops.Moment) -> bool:
-        """
-        Determines if it's possible to add an operation into a moment. An
+        """Determines if it's possible to add an operation into a moment. An
```
Comment (balopat) [inline cirq-core/cirq/neutral_atoms/neutral_atom_devices.py:260]: fix summary line

File: cirq-core/cirq/contrib/acquaintance/strategies/complete.py [hunk 30]
```diff
@@ -30,8 +30,7 @@
 def complete_acquaintance_strategy(
     qubit_order: Sequence['cirq.Qid'], acquaintance_size: int = 0, swap_gate: 'cirq.Gate' = ops.SWAP
 ) -> 'cirq.Circuit':
-    """
-    Returns an acquaintance strategy capable of executing a gate corresponding
```
Comment (balopat) [inline cirq-core/cirq/contrib/acquaintance/strategies/complete.py:34]: fix summary line

File: cirq-core/cirq/contrib/acquaintance/mutation_utils.py [hunk 31]
```diff
@@ -63,8 +63,7 @@ def replace_acquaintance_with_swap_network(
     acquaintance_size: Optional[int] = 0,
     swap_gate: 'cirq.Gate' = ops.SWAP,
 ) -> bool:
-    """
-    Replace every moment containing acquaintance gates (after
+    """Replace every moment containing acquaintance gates (after
```
Comment (balopat) [inline cirq-core/cirq/contrib/acquaintance/mutation_utils.py:66]: fix summary line

**Review by balopat — CHANGES_REQUESTED** [review]
Awesome! 

I added places where the summary line doesn't follow the style guide (I wish the linter would catch that!): 

https://google.github.io/styleguide/pyguide.html#381-docstrings - "...A docstring should be organized as a summary line (one physical line not exceeding 80 characters) terminated by a period, question mark, or exclamation point..."

**Comment by thanacles** [comment]
Thanks @balopat .  Since this is such a big change, I'm trying to keep all my changes trivial.  For the non-trivial cases, what do you think about me adding a TODO to update the docstring with a summary line?

**Comment by balopat** [comment]
> Thanks @balopat . Since this is such a big change, I'm trying to keep all my changes trivial. For the non-trivial cases, what do you think about me adding a TODO to update the docstring with a summary line?

Sounds good to me. A TODO + disabling pylint is better than a bad update.

File: cirq-core/cirq/neutral_atoms/neutral_atom_devices.py [hunk 32]
```diff
@@ -111,8 +107,7 @@ def duration_of(self, operation: ops.Operation):
         return self._gate_duration
 
     def validate_gate(self, gate: ops.Gate):
-        """
-        Raises an error if the provided gate isn't part of the native gate set.
+        """Raises an error if the provided gate isn't part of the native gate set.
```
[the remark this answers is not in the document]
Comment (thanacles) [reply cirq-core/cirq/neutral_atoms/neutral_atom_devices.py:110]: I'd argue that this is a good summary line for this validate method.  It's concern is to raise an error if it fails validation.  Also it has a Raises section below.

File: cirq-core/cirq/neutral_atoms/neutral_atom_devices.py [hunk 33]
```diff
@@ -141,8 +136,7 @@ def validate_gate(self, gate: ops.Gate):
                 raise ValueError('controlled gates must have integer exponents')
 
     def validate_operation(self, operation: ops.Operation):
-        """
-        Raises an error if the given operation is invalid on this device.
+        """Raises an error if the given operation is invalid on this device.
```
[the remark this answers is not in the document]
Comment (thanacles) [reply cirq-core/cirq/neutral_atoms/neutral_atom_devices.py:139]: Same as above.

File: cirq-core/cirq/neutral_atoms/neutral_atom_devices.py [hunk 34]
```diff
@@ -190,8 +184,7 @@ def validate_operation(self, operation: ops.Operation):
                 raise ValueError("Bad number of XY gates in parallel")
 
     def validate_moment(self, moment: ops.Moment):
-        """
-        Raises an error if the given moment is invalid on this device
+        """Raises an error if the given moment is invalid on this device
```
[the remark this answers is not in the document]
Comment (thanacles) [reply cirq-core/cirq/neutral_atoms/neutral_atom_devices.py:187]: ditto

[the remark this answers is not in the document]
Comment (thanacles) [reply cirq-core/cirq/neutral_atoms/neutral_atom_devices.py:187]: ditto

**Comment by thanacles** [comment]
@balopat Thanks again for bearing with me.  I think this lint rule will be nice once it's running implemented.  I've taken care of your suggestions, take a look.

File: cirq-core/cirq/neutral_atoms/neutral_atom_devices.py (hunk 33 above)
[the remark this answers is not in the document]
Comment (balopat) [reply cirq-core/cirq/neutral_atoms/neutral_atom_devices.py:139]: sounds good!

**Review by balopat — APPROVED** [review]
LGTM

**Pull request merged**